checkers/internal/lintutil/astflow.go

```diff
@@ -0,0 +1,88 @@
+package lintutil
+
+import (
+	"go/ast"
+	"go/token"
+	"go/types"
+
+	"github.com/go-toolsmith/astequal"
+	"github.com/go-toolsmith/astp"
+	"github.com/go-toolsmith/typep"
+)
+
+// Different utilities to make simple analysis over typed ast values flow.
+//
+// It's primitive and can't replace SSA, but the bright side is that
+// it does not require building an additional IR eagerly.
+// Expected to be used sparingly inside a few checkers.
+//
+// If proven really useful, can be moved to go-toolsmith library.
+
+// IsImmutable reports whether n can be midified through any operation.
+func IsImmutable(info *types.Info, n ast.Expr) bool {
+	if astp.IsBasicLit(n) {
+		return true
+	}
+	tv, ok := info.Types[n]
+	return ok && !tv.Assignable() && !tv.Assignable()
```

Comment:
```suggestion
	return ok && !tv.Assignable()
```

Reply:
Oh, it should be `!tv.Assignable() && !tv.Addressible()`